src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -0,0 +1,165 @@
+            : base(serviceProvider, stream, SpecializedCollections.EmptyEnumerable<JsonConverter>())
+        {
+            _languageName = languageName;
+
+            StartService();
+        }
+
+        [JsonRpcMethod(Methods.InitializeName)]
+        public object Initialize(int? processId, string rootPath, Uri rootUri, ClientCapabilities capabilities, TraceSetting trace, CancellationToken cancellationToken)
+        {
+            // our LSP server only supports WorkspaceStreamingSymbolProvider capability
+            // for now
+            return new InitializeResult()
+            {
+                Capabilities = new VSServerCapabilities()
+                {
+                    WorkspaceStreamingSymbolProvider = true
+                }
+            };
+        }
+
+        [JsonRpcMethod(Methods.InitializedName)]
+        public Task Initialized()
+        {
+            return Task.CompletedTask;
+        }
+
+        [JsonRpcMethod(Methods.ShutdownName)]
+        public void Shutdown(CancellationToken cancellationToken)
+        {
+            // our language server shutdown when VS shutdown
+            // we have this so that we don't get log file every time VS shutdown
+        }
+
+        [JsonRpcMethod(Methods.ExitName)]
+        public void Exit()
+        {
+            // our language server exit when VS shutdown
+            // we have this so that we don't get log file every time VS shutdown
+        }
+
+        [JsonRpcMethod(VSSymbolMethods.WorkspaceBeginSymbolName)]
+        public Task<VSBeginSymbolParams> WorkspaceBeginSymbolAsync(string query, int searchId, CancellationToken cancellationToken)
+        {
+            return RunServiceAsync(async () =>
+            {
+                using (UserOperationBooster.Boost())
+                {
+                    // for now, we use whatever solution we have currently. in future, we will add an ability to sync VS's current solution
+                    // on demand from OOP side
+                    // https://github.com/dotnet/roslyn/issues/37424
+                    await SearchAsync(SolutionService.PrimaryWorkspace.CurrentSolution, query, searchId, cancellationToken).ConfigureAwait(false);
+                    return new VSBeginSymbolParams();
+                }
+            }, cancellationToken);
+        }
+
+        private async Task SearchAsync(Solution solution, string query, int searchId, CancellationToken cancellationToken)
+        {
+            foreach (var project in solution.Projects.Where(p => p.Language == _languageName))
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var results = await AbstractNavigateToSearchService.SearchProjectInCurrentProcessAsync(
+                    project,
+                    ImmutableArray<Document>.Empty,
+                    query,
+                    s_supportedKinds,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
we appear to be losing our OOP parallelism here.

Reply:
this is new feature, so it never ran concurrently. @gundermanc do you want concurrent run? does VS side expects concurrent publish symbol calls?

Reply:
Client should be synchronized properly.

Reply:
so, you want us to report concurrently?

Reply:
>This is new feature, so it never ran concurrently.

The API you are calling was specifically and intentionally designed to run in parallel to get results as quickly as possible, while being really good about sharing progress and being efficient.  We perf tested the heck out of this and did lots and lots of work for this.  It doesn't make sense to me that we would start out here with this scenario not taking advantage of this (unless the core platform has an issue there).  

Thanks!

Reply:
well. whole Roslyn is designed to be run concurrently but we surely don't run everything concurrently. again, I think it is just picking words. I will leave it to @gundermanc what they want.

Reply:
I'm not sure i understand.  I **literally** wrote NavTo to be used in an **intentionally** concurrent fashion.  You're supposed to call it in parallel to get the best performance for the user.  That was a big part of moving OOP in the first place.  in proc we couldn't really do parallel because of the impact on the user.  But OOP there's no impact and you get nearly linear speedup with your core count.  If you don't do this, then running this on a project like Roslyn.sln will likely take N-x longer than getting the equiv results in NavTo.

Reply:
well. same thing repeated.

Reply:
alright. @gundermanc wants concurrency so, added that.